### PR TITLE
DEV: Migrate icons and links settings to new objects setting type

### PR DESCRIPTION
**Pull request by tgxworld**
### Why this change?

Since https://github.com/discourse/discourse/commit/a440e15291e98a866f80a6566d5ec0597ab77e2e, we have started to support objects typed theme setting so we are switching this theme component to use it instead as it provides a much better UX for configuring the settings required for the theme component.

### CODE REVIEW

File: .discourse-compatibility [hunk 1]
```diff
@@ -1,3 +1,4 @@
+< 3.3.0.beta2-dev: 0a837d7c044798417b1a7389cc37a67859c25dbe
```
Comment (nattsw) [inline .discourse-compatibility:1]: Where did you get this SHA from?

Comment (tgxworld) [reply .discourse-compatibility]: Ah good catch. I have no idea where that SHA came from but I have now updated the SHA to point to https://github.com/discourse/discourse-brand-header/commit/e64cfa531121f2cf152901f5b2a20512a5ee2c63

File: .discourse-compatibility [hunk 2]
```diff
@@ -1,3 +1,4 @@
+< 3.3.0.beta2-dev: e64cfa531121f2cf152901f5b2a20512a5ee2c63
```
Comment (tgxworld) [inline .discourse-compatibility:1]: https://github.com/discourse/discourse-brand-header/commit/e64cfa531121f2cf152901f5b2a20512a5ee2c63

File: settings.yaml [hunk 3]
```diff
@@ -6,16 +6,63 @@ logo_url:
...
+        required: true
+        validations:
+          min: 1
+          max: 2048
+          url: true
+      target:
+        type: enum
+        default: _blank
+        choices:
+          - _blank
```
Comment (nattsw) [inline settings.yaml:61]: Btw, why the `_`?

**Review by nattsw — APPROVED** [review]

File: settings.yaml (hunk 3 above)
Comment (tgxworld) [reply settings.yaml:61]: In `discourse-custom-header-links` it doesn't just use the target directly in the link's `target` attribute. 

https://github.com/discourse/discourse-custom-header-links/blob/6bd25fa7d740bd4888ddd8af3802feecc6e19bc2/javascripts/discourse/components/custom-header-links.js#L28

In the case of this plugin, the value of the property is used directly in the `target` attribute so we need to use the [actual valid values](https://www.w3schools.com/tags/att_a_target.asp)

**Comment by nattsw** [comment]
Screenshot 2024-04-24 at 11 28 31 AM

You cancelled the test?

**Comment by tgxworld** [comment]
Hmm nope, not sure what happened. I kicked off another rerun.

**Comment by tgxworld** [comment]
Thank you for reviewing @nattsw 👍

**Pull request merged**